src/System.Text.Encodings.Web/tests/HtmlEncoderTests.cs

```diff
@@ -14,6 +14,18 @@ namespace Microsoft.Framework.WebEncoders
     public class HtmlEncoderTests
     {
         [Fact]
+        public void TestSurrogate()
+        {
+            Assert.Equal("&#x1F4A9;", System.Text.Encodings.Web.HtmlEncoder.Default.Encode("\U0001f4a9"));
+            
+            using (var writer = new StringWriter())
+            {
+                System.Text.Encodings.Web.HtmlEncoder.Default.Encode(writer, "\U0001f4a9");
+                Assert.Equal("&#x1F4A9;", writer.GetStringBuilder().ToString());
+            }
```

Comment:
This test is mainly testing Surrogate characters with different scenario. so I prefer to keep this together